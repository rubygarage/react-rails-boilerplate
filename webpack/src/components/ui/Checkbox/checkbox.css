.checkbox {
  position: relative;
  padding-left: 30px;
  cursor: pointer;
}

.checkbox-input {
  display: none;
}

.checkbox-icon {
  $size: 2.4rem;

  position: absolute;
  left: 0;
  top: 0;
  display: inline-block;
  height: $size;
  width: $size;
  vertical-align: middle;
  margin-right: 1rem;
  line-height: $size;
  text-align: center;
  text-indent: 0;
  border-radius: 0.3rem;
  background-color: #9b9b9b;
  border: none;
  color: #4a4a4a;
}

.checkbox-text {
  font-size: 1.4rem;
}

.checkbox-input:checked + .checkbox-icon {
  border: none;
  background-color: #9b9b9b;

  &:before {
    content: '\e6a2';
    position: relative;
    right: -0.2rem;
  }
}

.vertical {
  text-align: center;
  flex-direction: column;
  text-indent: 0;
  padding-left: 0;

  .checkbox-icon {
    margin-right: 0;
    margin-bottom: 1rem;
  }

  .checkbox-text {
    display: block;
    font-size: 1.2rem;
    font-weight: 300;
  }
}

@media (min-width: 768px) {
  .checkbox-icon {
    $size: 1.8rem;

    height: $size;
    width: $size;
  }

  .checkbox-input:checked + .checkbox-icon:before {
    top: -0.5rem;
  }
}
