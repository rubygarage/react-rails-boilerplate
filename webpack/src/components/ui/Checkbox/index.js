import React, { Component } from 'react'
import { Field } from 'redux-form'
import classNames from 'classnames'

import styles from './checkbox.css'

class Checkbox extends Component {
  field = ({ input, className, name, align, text }) => (
    <label className={classNames(styles.checkbox, className, styles[align])}>
      <input {...input} type="checkbox" className={styles['checkbox-input']} name={name} />
      <span className={classNames(styles['checkbox-icon'], 'icon')} />
      <span className={styles['checkbox-text']}>{text}</span>
    </label>
  )

  render() {
    return <Field {...this.props} type="checkbox" component={this.field} />
  }
}

export default Checkbox
