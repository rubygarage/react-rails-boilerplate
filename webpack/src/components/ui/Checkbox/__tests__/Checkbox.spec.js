import React from 'react'
import { shallow } from 'enzyme'
import Checkbox from 'components/ui/Checkbox'

describe('<Checkbox />', () => {
  it('renders Checkbox', () => {
    const component = shallow(<Checkbox name="test" />)

    expect(component).toMatchSnapshot()
  })
})
