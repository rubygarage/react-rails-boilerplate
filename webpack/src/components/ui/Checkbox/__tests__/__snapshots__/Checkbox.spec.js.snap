// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders Checkbox 1`] = `
<Field
  component={[Function]}
  name="test"
  type="checkbox"
/>
`;
